var/spack/repos/builtin/packages/py-basis-set-exchange/package.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyBasisSetExchange(PythonPackage):
+    """Command-line interface for the Basis Set Exchange Website"""
+
+    homepage = "https://www.basissetexchange.org"
+    url      = "https://github.com/MolSSI-BSE/basis_set_exchange/archive/v0.8.12.tar.gz"
+
+    maintainers = ['bennybp', 'scemama']
+
+    version('0.8.12', sha256='41f0242e2c11392c511d3308c0345e14b0eb3344686d865bdfcb48257910fc0d')
+    version('0.8.11', sha256='dbd4cf9382d97b79a3143c6b2086d1d24562bd13e0ab0d18fc0423b9ee877b9b')
+    version('0.8.10', sha256='7f974faf513791d59ef47dd4eba9d8386f75bbd85f253ca5c08732ff56004a57')
+    version('0.8.9',  sha256='faf01a00c8ef7cf7331562052fbf1ff861e901545a927310ab09ac03604b400b')
+    version('0.8.8',  sha256='4770bc901b93bfbf1aa5e1c4b486804753b3ff19c41794d3200ffe990b12dfa1')
+    version('0.8.7',  sha256='5e9e18ca84bd378d1e02b6543cee122d8cc4efeaf05b1801061b500dda4cb2fc')
+    version('0.8.6',  sha256='96da4ab45adaab11825917dc4cf0e236f5d2f0af0f2e026578262ce389784912')
+    version('0.8.5',  sha256='ff8e6e03474319245e641c9da05715a6ea90dccab0e60ad2b0bc8556f85a9bca')
+    version('0.8.4',  sha256='005f95794ce55f1b1e8faa0b6910b814819dd4ee9dba367a3ec29dfe53816684')
+    version('0.8.3',  sha256='0721f3cf55f588f62d74a408bccdd44046ebeaab8ec802c02fae7983d8f0359f')
+
+    depends_on('py-argcomplete', type=('build', 'run'))
+    depends_on('py-jsonschema', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('python@3:', type=('build', 'run'))
+#    depends_on('py-versioneer', type='build')
```

Comment:
You definitely need this dependency. The beginning of `setup.py` imports versioneer, which won't be present in most Python installations.

Reply:
I don't believe versioneer needs to be installed prior to installing the package. I don't do that at all (for example, in the .travis.yml file)

What is being used is the versioneer.py file in the top level directory (next to setup.py).

Reply:
The way Spack installs Python packages is to run:
```console
$ python setup.py build
$ python setup.py install
```
Since the `setup.py` contains an import of `versioneer`, installation will fail without this dependency.

Reply:
Just tried it in a clean python environment and it works for me.

I would be very interested if it does fail for you/spack, since that means something in my setup is letting me get away with it, and it needs to be fixed on my end.

Reply:
Ah sorry, now I understand what you're saying. The `import versioneer` in `setup.py` imports the `versioneer.py` file in the same directory, not the `versioneer` library. It looks like this is a vendored copy of the library. In Spack we prefer not to use vendored copies of libraries if we can avoid it, but since it's only used at build-time, I don't think it matters. So we can remove this dependency.

Reply:
Yes, it should only be used at build time. Although to be honest, it does some fairly tricky things I am not fond of.

(Also, it appears to be a slightly-dead project now. I just haven't found a good alternative).